src/types.ts

```diff
@@ -0,0 +1,36 @@
+import { Domain } from "@zero-tech/data-store-core";
+
+export interface DomainsApiResponse {
```

Comment:
So I noticed that the DomainService either returns `Domain` or `DomainsApiResponse`

It seems odd that one gets called a `Response` and the other is just a `Domain`.

Why not drop the whole "Response" and "Api" part and just say what it's returning, `Domains`.

You could even type this
`export type Domains = Domain[]`

Reply:
The upcoming story to add typed responses is probably going to have to do this anyways. Ive just kept the response bodies similar to the original implementation